### PR TITLE
[fontconfig] Fix wrapper, usage

### PR DESCRIPTION
**Pull request by dg0yt**
- #### What does your PR fix?
  Fixes missing transitive usage requirements:
  `select_library_configurations(Fontconfig`) must be called before adding any other values to `Fontconfig_LIBRARES`.

  Adds explicit `usage` documentation.

### CODE REVIEW

File: ports/fontconfig/vcpkg-cmake-wrapper.cmake.in [hunk 1]
```diff
@@ -1,46 +1,47 @@
...
+        set_property(TARGET Fontconfig::Fontconfig APPEND PROPERTY IMPORTED_CONFIGURATIONS "Debug")
+        set_target_properties(Fontconfig::Fontconfig PROPERTIES IMPORTED_LOCATION_DEBUG "${Fontconfig_LIBRARY_DEBUG}")
+    endif()
     find_package(Freetype)
-    if(Freetype_FOUND AND TARGET Fontconfig::Fontconfig)
+    if(Freetype_FOUND)
+        list(APPEND Fontconfig_LIBRARIES "${FREETYPE_LIBRARIES}")
         if(TARGET Freetype::Freetype)
-            set_property(TARGET Fontconfig::Fontconfig APPEND PROPERTY INTERFACE_LINK_LIBRARIES "Freetype::Freetype")
+            set_property(TARGET Fontconfig::Fontconfig APPEND PROPERTY INTERFACE_LINK_LIBRARIES "\$<LINK_ONLY>:Freetype::Freetype>")
```
Comment (Neumann-A) [inline ports/fontconfig/vcpkg-cmake-wrapper.cmake.in:23]:
```suggestion
            set_property(TARGET Fontconfig::Fontconfig APPEND PROPERTY INTERFACE_LINK_LIBRARIES "\$<LINK_ONLY:Freetype::Freetype>")

```

File: ports/fontconfig/vcpkg-cmake-wrapper.cmake.in [hunk 2]
```diff
@@ -1,46 +1,47 @@
...
-        list(APPEND Fontconfig_LIBRARIES "${FREETYPE_LIBRARIES}")
-    endif()
     find_package(EXPAT)
-    if(EXPAT_FOUND AND TARGET Fontconfig::Fontconfig)
-        set_property(TARGET Fontconfig::Fontconfig APPEND PROPERTY INTERFACE_LINK_LIBRARIES "EXPAT::EXPAT")
-    endif()
-    if(EXPAT_LIBRARIES)
+    if(EXPAT_FOUND)
         list(APPEND Fontconfig_LIBRARIES "${EXPAT_LIBRARIES}")
+        set_property(TARGET Fontconfig::Fontconfig APPEND PROPERTY INTERFACE_LINK_LIBRARIES "\$<LINK_ONLY>:EXPAT::EXPAT>")
```
Comment (Neumann-A) [inline ports/fontconfig/vcpkg-cmake-wrapper.cmake.in:31]:
```suggestion
        set_property(TARGET Fontconfig::Fontconfig APPEND PROPERTY INTERFACE_LINK_LIBRARIES "\$<LINK_ONLY:EXPAT::EXPAT>")

```

Comment (Neumann-A) [reply ports/fontconfig/vcpkg-cmake-wrapper.cmake.in:31]: The EXPAT target is only available in `CMake 3.10`

**Comment by dg0yt** [comment]
@Adela0814 It is for a "port-bug", not a feature.

**Review by Adela0814 — APPROVED** [review]

**Pull request merged**